### DISCUSSION TITLE
UI for enabling plugins or assigning them to "extension points" is unclear and needs design

### DISCUSSION THREAD

**Discussion by cbjeukendrup** [musescore/MuseScore#24107] [Ideas]
### Issue type

UI bug (incorrect info or interface appearance)

### Bug description

In the panel to manage a plugin's shortcut and enabledness, there is now a dropdown to enable a plugin or assign it to an "extension point".

This will need a UI design pass. The main problems with it are the following:
- It isn't clear that "Manually" is the new "Enabled"
- The other options are badly worded ("On post/pre …" is not correct English)
- The user has no way to understand what these options do. 

Also, I'm personally still of the opinion that this is not the right way to present these "extension points" to the user. Instead, I'd prefer to see it as a sort of "if this than that" style automation, where you see a list of possible triggers and then assign plugins or other actions to these triggers. Now, it feels like it's the wrong way around, where you have to assign a trigger to a plugin.

### Steps to reproduce

1. Open Home > Plugins
2. Click a plugin to see the panel appear

### Screenshots/Screen recordings

<img width="1440" alt="Scherm­afbeelding 2024-07-01 om 16 56 13" src="https://github.com/musescore/MuseScore/assets/48658420/47d3ee0f-02c0-4568-9e84-40c844e65f44">

### MuseScore Version

master (4.4)

### Regression

Yes, this used to work in a previous version of MuseScore 4.x

### Operating system

\*

**Comment by TeeDeeY** [comment]
A related issue is that setting title: “title” conditionally in version 4.4 in the “onCompleted” section no longer works.  The Jan 4 version has the changes described here and title in the menu doesn’t work, conditionally.   Late June 4.4 does work.  

Since I don’t know the design goals, it’s difficult to comment.  But, using conditional logic allows one plug-in to load in versions 3.6, 4.2 and 4.4, assuming other components are also compatible.   

If this should be reported and documented separately, please tell me.  It could be that titles go away by design, making posting pointless.

**Comment by cbjeukendrup** [comment]
That is more related to https://github.com/musescore/MuseScore/issues/23419, but I think it would be good to open a new issue for that. 
The idea is not to remove the concept of plugin titles, but the old API will be deprecated at some point. 
However, for 4.4 there shouldn't really be any changes. The reason that you're still seeing some regressions in the latest nightly builds, is that development of the new API has already started. We'll try our best to minimise the amount of regressions before the final 4.4 release. 
(An official announcement about the developments around the plugins API will follow soon.)

**Comment by bkunda** [comment]
This was reverted by @igorkorsukov so it no longer appears in 4.4. 
We'll be working on a proper design solution to present these triggers to users for 4.5.

**Comment by cbjeukendrup** [comment]
Why was this converted to a discussion? That means we can't really do anything useful with this anymore...
